trino/dbapi.py

```diff
@@ -373,10 +373,11 @@ def _prepare_statement(self, statement: str, name: str) -> None:
         :param statement: sql to be executed.
         :param name: name that will be assigned to the prepared statement.
         """
-        sql = f"PREPARE {name} FROM {statement}"
-        query = trino.client.TrinoQuery(self.connection._create_request(), query=sql,
-                                        legacy_primitive_types=self._legacy_primitive_types)
-        query.execute()
+        # sql = f"PREPARE {name} FROM {statement}"
+        # query = trino.client.TrinoQuery(self.connection._create_request(), query=sql,
+        #                                 legacy_primitive_types=self._legacy_primitive_types)
+        # query.execute()
+        self.connection._client_session.prepared_statements[name] = statement
```

Comment:
if we want to reduce http calls the alternative is to implement client side parameter handling and not use server side prepared statements at all (which I'm not a fan of).

An option is to remove the `DEALLOCATE` call and do a deallocate for all tracked statements during connection close. That has the benefit that you can avoid a `PREPARE` as well for already prepared statements (since we know the already prepared statements on the client side).

However this entire thing feels like a workaround.

Reply:
I know this is Python, but is there some concurrency risk by modifying `_client_session`?